cmd/settings.go

```diff
@@ -0,0 +1,223 @@
+		Short:        "List and manage settings",
+		Long:         `List and manage settings`,
+		SilenceUsage: true,
+	}
+
+	cmd.AddCommand(settingShowCmd())
+	cmd.AddCommand(settingUpdateCmd())
+	cmd.AddCommand(schemaListCmd())
+	cmd.AddCommand(schemaShowCmd())
+
+	return cmd
+}
+
+//
+//
+func settingShowCmd() *cobra.Command {
+	options := settingsOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Show settings for a specific scope/section",
+		Long:  `Show settings for a specific scope/section. Scope is by default GLOBAL.`,
+		Example: `
+	privx-cli settings show [access flags] --scope <SCOPE>
+	privx-cli settings show [access flags] --scope <SCOPE> --section <SECTION>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return settingShow(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&options.scope, "scope", "", "GLOBAL", "scope setting name")
+	flags.StringVar(&options.section, "section", "", "section setting name")
+	flags.StringVar(&options.merge, "merge", "", "signal whether service specific settings should be merged with shared settings. Compatible with scope settings only")
+
+	return cmd
+}
+
+func settingShow(options settingsOptions) error {
+	api := settings.New(curl())
+
+	if options.section != "" {
+		if options.merge != "" {
+			fmt.Fprintln(os.Stderr, "Error: --merge flag is compatible with scope settings only")
+			os.Exit(1)
+		}
+
+		res, err := api.ScopeSectionSettings(strings.ToUpper(options.scope),
+			strings.ToLower(options.section))
+		if err != nil {
+			return err
+		}
+
+		return stdout(res)
+	} else {
+		res, err := api.ScopeSettings(strings.ToUpper(options.scope), options.merge)
+		if err != nil {
+			return err
+		}
+
+		return stdout(res)
+	}
+}
+
+//
+//
+func settingUpdateCmd() *cobra.Command {
+	options := settingsOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update scope/section settings",
+		Long:  `Update scope/section settings. Scope is by default GLOBAL.`,
+		Example: `
+	privx-cli settings update [access flags] --scope <SCOPE> JSON-FILE
+	privx-cli settings update [access flags] --scope <SCOPE> --section <SECTION> JSON-FILE
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return settingUpdate(options, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&options.scope, "scope", "", "GLOBAL", "scope setting name")
+	flags.StringVar(&options.section, "section", "", "section setting name")
+
+	return cmd
+}
+
+func settingUpdate(options settingsOptions, args []string) error {
+	var updateSettings json.RawMessage
+	api := settings.New(curl())
+
+	err := decodeJSON(args[0], &updateSettings)
+	if err != nil {
+		return err
+	}
+
+	if options.section != "" {
+		err = api.UpdateScopeSectionSettings(&updateSettings, strings.ToUpper(options.scope),
+			strings.ToLower(options.section))
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
I personally prefer switch and, in this case, delay the processing of err until exiting the switch scope. Since the operation is just one liner this is hopefully easier to grasp (remember cognitive complexity for nested conditionals and loops!)

```
switch options.section {
    case "":
        err = api.do_something(args)
    default:
        err = api.do_something_else(args)
}

// Passing the err along and not adding any more debug info to it
return err
```